src/main/groovy/com/bmuschko/gradle/docker/utils/TaskStateHelper.groovy

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2014 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.bmuschko.gradle.docker.utils
+
+import org.gradle.api.Project
+
+class TaskStateHelper {
+
+    String taskName
+
+    Project project
+
+    public TaskStateHelper(String taskName, Project project) {
```

Comment:
Instead of passing in the Project instance, let's just pass in `File buildDir`. That'll decouple the Gradle API from this class and allows us to write unit tests. Let's also write a unit test or turn the existing on into a unit test.